Extension/lib/content-message-handler.js

```diff
@@ -457,11 +457,23 @@
                 adguard.sync.syncService.setSyncProvider(message.provider);
                 break;
             case 'setOauthToken':
+                adguard.sync.oauthService.setToken(message.provider, message.token, message.securityToken, message.expires);
                 adguard.sync.syncService.setSyncProvider(message.provider, message.token, message.securityToken, message.expires);
                 break;
             case 'onAuthError':
                 adguard.sync.syncService.removeSyncProvider(message.provider);
                 break;
+            case 'getSyncStatus':
+                return adguard.sync.syncService.getSyncStatus();
+            case 'authSync':
+                adguard.sync.syncService.authorize();
+                break;
+            case 'toggleSync':
+                adguard.sync.syncService.toggleSyncStatus();
+                break;
+            case 'syncNow':
+                adguard.sync.syncService.syncSettings();
```

Comment:
I think it's ok to provide the option, actually in this case it just goes into queue.